crates/types/src/nodes_config.rs

```diff
@@ -57,15 +60,84 @@ pub enum Role {
 #[display(fmt = "v{}", _0)]
 pub struct ConfigVersion(u32);
 
+impl ConfigVersion {
+    pub const INVALID: ConfigVersion = ConfigVersion(0);
+    pub const MIN: ConfigVersion = ConfigVersion(1);
+}
+
+impl Default for ConfigVersion {
+    fn default() -> Self {
+        Self::MIN
+    }
+}
+
 #[derive(Debug, Clone, Eq, PartialEq, Default)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-struct NodesConfiguration {
+pub struct NodesConfiguration {
     version: ConfigVersion,
     nodes: HashMap<PlainNodeId, MaybeNode>,
     controllers: Vec<ControllerConfig>,
     name_lookup: HashMap<String, PlainNodeId>,
 }
 
+impl NodesConfiguration {
+    /// The currently loaded nodes configuration. This returns None if no configuration
+    /// is loaded yet, this is will be only the case before attaching to the controller
+    /// node on startup.
+    ///
+    /// After attaching, it's safe to unwrap this, or use current_unchecked() variant.
+    pub fn current() -> Option<Arc<NodesConfiguration>> {
+        CURRENT_CONFIG.load_full()
+    }
+
+    pub fn current_unchecked() -> Arc<NodesConfiguration> {
+        CURRENT_CONFIG.load_full().unwrap()
+    }
+
+    /// The currently loaded configuration version or ConfigVersion::INVALID if
+    /// no configuration is loaded yet.
+    pub fn current_version() -> ConfigVersion {
+        CURRENT_CONFIG
+            .load()
+            .as_ref()
+            .map(|config| config.version)
+            .unwrap_or(ConfigVersion::INVALID)
+    }
+
+    /// Silently ignore the config if it's older than existing
+    pub fn set_as_current_if_newer(self) -> ConfigVersion {
```

Comment:
What does the return value represent? From the code it looks that it is the config version of the previous configuration (independent of whether it was overwritten or not). Should it always return the version of the configuration which is now loaded into the `ArcSwap`?

Reply:
It's actually quite useless, will remove.